pytorch_lightning/utilities/argparse.py

```diff
@@ -232,7 +232,6 @@ def add_argparse_args(
 
         if arg == 'gpus' or arg == 'tpu_cores':
             use_type = _gpus_allowed_type
-            arg_default = _gpus_arg_default
 
```

Comment:
this is the fix.
we should not attach a ruction, because if we pass it into our hparams, it will land in the checkpoint and risk not being able to unpickle.